samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -134,7 +134,12 @@
   public void start() {
     startMetrics();
     systemAdmins.start();
-    zkController.register();
+    ZkKeyBuilder keyBuilder = zkUtils.getKeyBuilder();
+    zkUtils.validateZkVersion();
+    zkUtils.validatePaths(new String[]{keyBuilder.getProcessorsPath(), keyBuilder.getJobModelVersionPath(), keyBuilder
```

Comment:
Do these validations throw? If so, does that result in a call to stop? If not, maybe close systemAdmins in a try/finally, or start them after these validations.

Reply:
Minor, Formatting: maybe move the newline earlier instead of the last method call?